hooks.go

```diff
@@ -0,0 +1,113 @@
+package easypost
+
+import (
+	"context"
+	"github.com/google/uuid"
+	"io"
+	"net/url"
+	"time"
+)
+
+// HookEvent is the base type for all hook events
+type HookEvent struct {
+}
+
+// HookEventSubscriber is the base type for all hook event subscribers.
+// ID needs to be unique in order to find and remove the subscriber
+type HookEventSubscriber struct {
+	ID string // need unique ID for removal
+}
+
+// RequestHookEvent is the data passed to a RequestHookEventSubscriberCallback function
+type RequestHookEvent struct {
+	HookEvent        // implements HookEvent
```

Comment:
I am not sure why we need this

Reply:
This is the data object that will be passed to the callback function, like we did in Java and .NET